python/cog/server/runner.py

```diff
@@ -381,16 +394,25 @@ def shutdown(self) -> None:
         if self._child.is_alive():
             self.log.info("child is alive during shutdown, sending Shutdown event")
             self._events.send(Shutdown())
+            asyncio.create_task(self.terminate_later())
```

Comment:
This doesn't make sense to me. Generally in this codebase `shutdown` has meant "graceful shutdown" and `terminate` has meant "stop what you're doing right now". It doesn't really make sense for the former to call the latter.

It's also more than a little hairy for this kind of stuff to be coordinated in `Worker` rather than at the top level of the program as it is in `main` (see my comment above).